### DISCUSSION TITLE
Can I get windows LEMto use ccl for the lisp-repl?

### DISCUSSION THREAD

**Discussion by dsacks** [lem-project/lem#749] [General]
I want to try to use ccl for LEM, but I'm not sure how.

I put this in my init.lisp:

(lem-lisp-mode:start-lisp-repl t)

but I'm getting sbcl.

**Comment by jcs090218** [comment]
The CCL implementation is in https://github.com/lem-project/lem/pull/686.

**Comment by Sasanidas** [comment]
> I want to try to use ccl for LEM, but I'm not sure how.
> 
> I put this in my init.lisp:
> 
> (lem-lisp-mode:start-lisp-repl t)
> 
> but I'm getting sbcl.

Sorry I wasn't aware of the Github discussion pages, as @jcs090218 mention, I've been working (here https://github.com/lem-project/lem/pull/686) to support multiple CL implementations. 

If you want to use CCL, feel free to download that branch (or apply the diff to the main branch) and run lem normally, using the ccl repl:
```
(ql:quickload :lem-ncurses)

(lem:lem)
```

For now, in CCL only the ncurses fronted has been tested, and still have a few visual bugs (as mention here https://github.com/lem-project/lem/pull/686#issuecomment-1587219292).

**Reply by dsacks** [reply]
This is maybe more a CCL question or I don't even know, but when I try to quickload lem-ncurses I get this error about sb-concurrency not being found - and not had much luck goggling sb-concurrency.

? (ql:quickload :lem-ncurses)
To load "lem-ncurses":
  Load 1 ASDF system:
    lem-ncurses
; Loading "lem-ncurses"
.> Error: System "sb-concurrency" not found
> While executing: (:INTERNAL QUICKLISP-CLIENT::RECURSE QUICKLISP-CLIENT::COMPUTE-LOAD-STRATEGY), in process listener(1).

> Type :GO to continue, :POP to abort, :R for a list of available restarts.
> If continued: Try again
> Type :? for other options.

**Reply by Sasanidas** [reply]
Umm, if you try with the main branch or the 2.0 executable is not going to work, did you try with the branch https://github.com/lem-project/lem/pull/686?

**Reply by dsacks** [reply]
I'm having a hard time finding branch 686.

**Reply by dsacks** [reply]
Uh, I checked out origin/other-lisp-implementations and tried again and got further:

; Loading "lem-ncurses"                                                                                                 
..................................................                                                                      
[package cl-charms/low-level]; gcc -o C:/Users/dennis/AppData/Local/cache/common-lisp/ccl-1.12-f98-win-x64/C/Users/denni
s/quicklisp/dists/quicklisp/software/cl-charms-20221106-git/src/low-level/curses-grovel__grovel-tmpQH0RTDWG.obj -c -m64 
-IC:/Users/dennis/quicklisp/dists/quicklisp/software/cffi-20230214-git/ C:/Users/dennis/AppData/Local/cache/common-lisp/
ccl-1.12-f98-win-x64/C/Users/dennis/quicklisp/dists/quicklisp/software/cl-charms-20221106-git/src/low-level/curses-grove
l__grovel.c                                                                                                             
> Error: Subprocess #<UIOP/LAUNCH-PROGRAM::PROCESS-INFO #x210188063D>                                                   
>         with command ("gcc" "-o" "C:/Users/dennis/AppData/Local/cache/common-lisp/ccl-1.12-f98-win-x64/C/Users/dennis/
quicklisp/dists/quicklisp/software/cl-charms-20221106-git/src/low-level/curses-grovel__grovel-tmpQH0RTDWG.obj" "-c" "-m6
4" "-IC:/Users/dennis/quicklisp/dists/quicklisp/software/cffi-20230214-git/" "C:/Users/dennis/AppData/Local/cache/common
-lisp/ccl-1.12-f98-win-x64/C/Users/dennis/quicklisp/dists/quicklisp/software/cl-charms-20221106-git/src/low-level/curses
-grovel__grovel.c")                                                                                                     
>         exited with error code 1                                                                                      
> While executing: CFFI-GROVEL:GROVEL-ERROR, in process listener(1).                                                    
> Type :POP to abort, :R for a list of available restarts.                                                              
> Type :? for other options.                                                                                            
1 >

**Reply by Sasanidas** [reply]
Right, I didn't test i on Windows, it maybe that you have to do some  extra steps to compile cl-charms, you can find more information here (https://github.com/HiTECNOLOGYs/cl-charms#windows)

**Reply by dsacks** [reply]
What do you run under? Linux? MacOS?

**Reply by Sasanidas** [reply]
I'm in Linux Mint Debian version

**Reply by cxxxr** [reply]
https://github.com/lem-project/lem/wiki/Windows-Platform
Would this page be helpful?
